src/isomorphism.rs

```diff
@@ -411,14 +432,48 @@ where
     }
 
     let g0 = if !id_order {
-        inner_temp_g0 = Vf2ppSorter.reorder(py, g0_out);
+        inner_temp_g0 = if mapping.is_some() {
+            let mut vf2pp_map: HashMap<usize, usize> =
+                HashMap::with_capacity(g0_out.node_count());
+            let temp = Vf2ppSorter.reorder(py, g0_out, Some(&mut vf2pp_map));
+            match node_map_g0 {
+                Some(ref mut g0_map) => {
+                    let mut temp_map = HashMap::with_capacity(g0_map.len());
+                    for (new_index, old_index) in g0_map.iter_mut() {
+                        temp_map.insert(vf2pp_map[&new_index], *old_index);
+                    }
+                    *g0_map = temp_map;
```

Comment:
```suggestion
                    for (_, old_index) in vf2pp_map.iter_mut() {
                        *old_index = g0_map[old_index];
                    }
                    *g0_map = vf2pp_map;
```
here we should just change the values of `vf2pp_map`  using `g0_map` to go back to the original graph indices.